fastly/resource_fastly_integration.go

```diff
@@ -0,0 +1,149 @@
+package fastly
+
+import (
+	"context"
+	"log"
+
+	gofastly "github.com/fastly/go-fastly/v9/fastly"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/diag"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+)
+
+func resourceFastlyIntegration() *schema.Resource {
+	return &schema.Resource{
+		CreateContext: resourceFastlyIntegrationCreate,
+		ReadContext:   resourceFastlyIntegrationRead,
+		UpdateContext: resourceFastlyIntegrationUpdate,
+		DeleteContext: resourceFastlyIntegrationDelete,
+		Importer: &schema.ResourceImporter{
+			StateContext: schema.ImportStatePassthroughContext,
+		},
+
+		Schema: map[string]*schema.Schema{
+			"config": {
+				Type:        schema.TypeMap,
+				Required:    true,
+				Description: "Configuration specific to the integration `type`.",
```

Comment:
This might seem a bit silly but I'm very used to users not reading the docs and just clicking through to the schema and missing any examples we've written, so in this case, as `config` is tightly coupled to `type` (i.e. if a user reads the description alone, will they know what value to provide) I think this additional comment might be useful...

```suggestion
				Description: "Configuration specific to the integration `type` (see documentation examples).",
```